### ISSUE TITLE
`position_embeddings` argument error when running in eval() mode

### ISSUE THREAD

**Issue by MinkyoungCho** [cornstarch-org/Cornstarch#52]
I encountered an error when running the model in `model.eval() `mode that does not appear in `model.train()` mode. The error is related to the position_embeddings argument.

```
model.eval()
with torch.no_grad():
    with tqdm(
        range(total_step),
        desc=f"Validation",
    ) as pbar:
        for item in pbar:
            inputs = next(dataload_iter)
            outputs = model(**inputs)
```

I am currently running a VLM (SigLIP + Qwen2).

In more detail, the model call chain looks like below. 
In `cornstarch/models/multimodal_language_model/modeling_multimodal_language_model.py:`

```
In class MultimodalModel(nn.Module):
     ...
    return self.language_model(**language_model_inputs) ###  this eventually calls function in `modeling_qwn2.py`
```

In `<python site-packages path>/transformers/models/qwen2/modeling_qwen2.py, at line 579` (see below), I get the following error when in eval mode: `got multiple values for keyword argument 'position_embeddings'` error (refer to the snapshot below),

```
layer_outputs = decoder_layer(
                     hidden_states,
                    attention_mask=causal_mask,
                    position_ids=position_ids,
                    past_key_value=past_key_values,
                    output_attentions=output_attentions,
                    use_cache=use_cache,
                    cache_position=cache_position,
                    position_embeddings=position_embeddings,
                    **flash_attn_kwargs,
                )
```

<img width="1022" alt="Image" src="https://github.com/user-attachments/assets/ad21b04b-d12b-4e75-9699-936cb1cab910" />

Currently, position_embeddings has a tuple format, which doesn't seem unusual. Do you have any idea why this error only occurs in `eval()` mode and not in `train()` mode?

**Comment by MinkyoungCho** [comment]
+ in `<python site-packages path>/transformers/models/qwen2/modeling_qwen2.py`, I found this.

```
for decoder_layer in self.layers[: self.config.num_hidden_layers]:
            if output_hidden_states:
                all_hidden_states += (hidden_states,)

            if self.gradient_checkpointing and self.training:
                layer_outputs = self._gradient_checkpointing_func(
                    decoder_layer.__call__,
                    hidden_states,
                    causal_mask,
                    position_ids,
                    past_key_values,
                    output_attentions,
                    use_cache,
                    cache_position,
                    position_embeddings,
                )
            else:
                layer_outputs = decoder_layer(
                    hidden_states,
                    attention_mask=causal_mask,
                    position_ids=position_ids,
                    past_key_value=past_key_values,
                    output_attentions=output_attentions,
                    use_cache=use_cache,
                    cache_position=cache_position,
                    position_embeddings=position_embeddings,
                    **flash_attn_kwargs,
                )
```

The code path diverges depending on the mode: in model.train() mode, the `if` branch is executed, while in eval() mode, the `else` branch is taken.
